### ISSUE TITLE
how to generate CAM?

### ISSUE THREAD

**Issue by aoyanl** [GuoleiSun/MCIS_wsss#9]
att_path = './runs2/exp1_mcis/att_clear_epoch10/'   where is the localization map saved ？
I have some questions about the following code. first, what is the index? and why atts=self.map1,not self.map2 ?  Second,what is the meaning of the item "batch_index+index" of the accu_map_name ? 
I really hope you can answer my question. Thank you ！
CODE：
 if index != None and epoch >= 0:
            atts = self.map1
            atts[atts < 0] = 0
            ind = torch.nonzero(label)
            for i in range(ind.shape[0]):  
                batch_index, la = ind[i]   
                accu_map_name = '{}/{}_{}.png'.format(self.att_dir, batch_index+index, la)
                
                att = atts[batch_index, la].cpu().data.numpy()
                att = att / (att.max() + 1e-8) * 255
                
                # if this is the last epoch and the image without any accumulation
                if epoch == self.training_epoch - 1 and not os.path.exists(accu_map_name):
                    cv2.imwrite(accu_map_name, att)
                    continue
                
                #naive filter out the low quality attention map with prob
                if probs[batch_index, la] < 0.1:  
                    continue

                if not os.path.exists(accu_map_name):
                    cv2.imwrite(accu_map_name, att)
                else:
                    accu_att = cv2.imread(accu_map_name, 0)
                    accu_att = np.maximum(accu_att, att)
                    cv2.imwrite(accu_map_name,  accu_att)

**Comment by GuoleiSun** [comment]
Hello, 
Thanks for your interest. When you finish the training, you have to run "./test.sh" to generate the localisation map. In "test.sh", you can set the att_path for saving localization map. For the code you mentioned, we didn't use it to generate localization map. Please check the corresponding arguments.
